tests/upgrade_path/test_upgrade_path.py

```diff
@@ -246,6 +246,16 @@ def test_upgrade_path(localhost, duthosts, rand_one_dut_hostname, ptfhost, upgra
                         platform="remote",
                         qlen=10000,
                         log_file=log_file)
+
+            ctr = 120
+            # Poll every 5s for 10min
+            while ctr:
+                reboot_cause = get_reboot_cause(duthost)
+                if reboot_cause == upgrade_type:
+                    break
+                else:
+                    time.sleep(5)
+                ctr -= 1
```

Comment:
I don't understand why it would be necessary to poll for 10 minutes here.

However, if this _is_ determined to be necessary, I would prefer a more understandable timer. `ctr = 120` and then subtracting 1 every 5 seconds isn't obvious at first glance. I'd prefer obtaining the start time before entering the loop and at the top of each loop doing something like:

```python
start_secs = time.time()
while (time.time() - start_secs) < 600:
    ...
```

Also, why are we getting the reboot cause again on line 259?

Reply:
while testing fast-boot, the test looks for reboot-cause immediately after the fast-boot and the correct reboot-cause is not available the time. This polling is basically retrying for every 5s for 10min. If found earlier, it will stop polling. We just have an upper bound of 10min.

Reply:
Rather than give an arbitrary timeout, can we instead wait until the `process-reboot-cause` service has finished running?